**RAHUL KHARE**

**52010079**

**MCA 4TH SEM**

**DATA ANALYTICS ASSIGNMENT**

**Solving System of Linear Equations**

--- 

Consider the following system of linear equations:

x + 2y = 0

1.5x -y = 0.5

So, to solve the above system of linear equations, we need to find the values of x and y

So, these equations can be represented as matrix form by AX = B

here  **Matrix A is**

matrix A = $\begin{bmatrix}
1 & 2 \\
1.5 & -1 \\
\end{bmatrix}$

and is defined as a list:

```
A = [[1.0, 2.0],[1.5,-1.0]]
```

or as a numpy array:

```python
A = np.array([[1.0, 2.0],[1.5,-1.0]])
```

X = $\begin{bmatrix}
x \\ y
\end{bmatrix}$

**And Vector B is**

B = $\begin{bmatrix}
0 \\ 0.5
\end{bmatrix}$

So, to find the value of x and y, we need to find the values of the matrix X. To do so, we can take the dot product of the inverse of matrix A, and the matrix B

X = inverse(A).B

i.e  $X=A^{-1}\,B$

This can be done in following way :-

In [ ]:
!pip install numpy

In [ ]:
import numpy as np

In [ ]:
A = np.array([[1, 2], [1.5, -1,]])
print(A)

[[ 1.   2. ]
 [ 1.5 -1. ]]


In [ ]:
b = np.array([0,0.5])

Calculating the inverse of matrix A

In [ ]:
invA = np.linalg.inv(A)

In [ ]:
print(invA)

[[ 0.25   0.5  ]
 [ 0.375 -0.25 ]]


Calculating the Dot product with matrix B

In [ ]:
x = np.dot(invA,b)

Printing Result/Answer

In [ ]:
print(x)

[ 0.25  -0.125]


So, the solution to the above equations is x = 0.25 and y = -0.125 which satisfies the System of linear equations.

# Example 2 : 

Let's solve a system of three linear equations which are as follows:

x + y + z = 6

x -y + z = 2

2x - y + 3z = 9



In [ ]:
A = np.array([[1, 1, 1], [1, -1, 1], [2, -1, 3]])
B = np.array([6, 2, 9])

Finding inverse of matrix A and multiplying it with matrix B

In [ ]:
invA = np.linalg.inv(A)
x = np.dot(invA,B)
print(x)

[1. 2. 3.]


so, x=1  |  y=2  |  z=3  is the correct solution and satisfies the System of linear equations.



Instead of using linalg.inv() and linalg.dot() two separate methods, we can do this in a single step by linalg.solve()

In [ ]:
A = np.array([[1, 1, 1], [1, -1, 1], [2, -1, 3]])
B = np.array([6, 2, 9])
X2 = np.linalg.solve(A,B)

print(X2)

[1. 2. 3.]


again, we got the same answer using linalg.solve() method also i.e x=1 | y=2 | z=3



---





